packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -250,6 +268,34 @@ public void onSnapshotReady(Bitmap bitmap) {
           markersController.changeMarkers(markersToChange);
           List<Object> markerIdsToRemove = call.argument("markerIdsToRemove");
           markersController.removeMarkers(markerIdsToRemove);
+
+          // gmscore GL renderer uses a TextureView.
+          // Android platform views that are displayed as a texture after Flutter v3.0.0.
+          // require that the view hierarchy is notified after all drawing operations have been flushed.
+          // Since the GL renderer doesn't use standard Android views, and instead uses GL directly,
+          // we notify the view hierarchy by invalidating the view.
+          // Unfortunately, when OnMapLoadedCallback is fired, the texture may not have been updated yet.
+          // To workaround this limitation, wait two frames.
+          // This ensures that at least the frame budget (16.66ms at 60hz) have passed since the
+          // drawing operation was issued.
+          if (googleMap != null) {
+            googleMap.setOnMapLoadedCallback(
+                new GoogleMap.OnMapLoadedCallback() {
+                  @Override
+                  public void onMapLoaded() {
+                    postFrameCallback(
+                        () -> {
+                          postFrameCallback(
+                              () -> {
+                                if (mapView != null) {
+                                  mapView.invalidate();
+                                }
+                              });
+                        });
+                  }
+                });
+          }
```

Comment:
Since this is a block of specialized, non-trivial logic, I would definitely suggest putting it in a helper function. Then the comment about how it works can go there, and this can become just a call to that helper with a comment explaining why it's only done in `markers#update` (since it's definitely non-obvious).

That will also make it trivial to add this to other codepaths if we discover other edge cases later that have this same issue.

Reply:
done